packages/ngx/package.json

```diff
@@ -0,0 +1,145 @@
+{
+	"name": "@gov.au/ngx",
+	"version": "0.6.5",
+	"description": "Angular implementation of GOV.AU UI-Kit modules using CDK",
+	"keywords": [
+		"uikit",
+		"pancake",
+		"pancake-module",
+		"sass",
+		"scss",
+		"css",
+		"gov.au",
+		"Commonwealth of Australia",
+		"design guide",
+		"design system",
+		"Angular",
+    "Angular Component Development Kit",
+    "Angular CDK",
+		"Angular module"
+	],
+	"scripts": {
+		"test": "ng test --sourcemaps=false",
+		"lint": "ng lint -fix",
+		"format": "prettier --write 'src/**/*.ts'",
+		"prepublish": "npm run test && npm run build",
+		"build": "ng-packagr -p package.json",
+		"watch": "ng serve --extract-css"
+	},
+  "pre-commit": [
+    "format",
+    "test"
+  ],
+	"dependencies": {
+		"@gov.au/core": "^2.0.0",
+		"@angular/core": "^5.2.0",
+		"@gov.au/accordion": "^2.0.0",
+		"@gov.au/body": "^2.0.0",
+		"@gov.au/link-list": "^2.0.0"
+	},
+	"peerDependencies": {},
```

Comment:
dependencies have to be mirrored inside peerDependencies for [pancake](https://github.com/govau/pancake) to work

Reply:
@dominikwilkowski on your points ..

The way we have used this mono repo is by adding each entry point for each framework to each component rather than creating new ones. That way we can reuse css etc. The way you added this in one monolithic chuck breaks this unfortunately

- I thought about doing the same but it would have doubled up on the packages for little benefit. Let's not even mention that there will be an angular cli build project in each of the libraries. So I have taken angular material approach - one project with multiple exportable library packages. Have a look at https://github.com/angular/material2.

- This library does does not create alternative CSS for the other libraries, rather it requires those CSS files to be included manually (https://github.com/shakilsiraj/uikit/blob/feature-angular/packages/ngx/src/sass/_module.scss) into Angular projects that use this library. So, I don't think it's an issue.

Introducing several new ways of building this is probably not the best strategy if you do that only in one single component. If we want to move to karma/jasmine we should do that everywhere not in one component and end up with several testing frameworks that all do the same. I for one would suggest staying with Jest and adding cypress for e2e tests to cover everything.

- If you are going to support Angular cli based projects then there is no other way but to go with  Karma/Jasmine.  A single project with this exception(not using Jest and use the appropriate libraries that are required) will keep it under control. Jest is a good framework no doubt but it's an added library that you will need to support on top of what Angular is providing. In Angular world, you can try to not use Karma/Jasmin, but you will be doing extra work and will need to maintain that work for yourself.


typescript, while awesome and amazing, is also something we should have a discussion about. I like typed languages but should probably pick one that does not diverge from the standard so we're not locked into an ecosystem (typescript has a commenting syntax that would help here)
- Angular does not support vanilla JS (directives, enums, proper OOP, etc. are missing there). So, there is not alternative.


Also in general a PR with 83 files changed is probably a bit much to be merged unless clearly communicated before hand with the team. I do however totally appreciate the effort you put into this.
- Thanks for that and sorry for not communicating before hand. Most of those files in this PR would be standard from angular CLI when you start a project. The bit that you should be focusing on is https://github.com/shakilsiraj/uikit/tree/feature-angular/packages/ngx/src/app/modules. Next PRs from anyone should be small.

Happy to discuss.